pkg/cmd/pr/merge/merge.go

```diff
@@ -316,12 +345,48 @@ func deleteBranchSurvey(opts *MergeOptions, crossRepoPR bool) (bool, error) {
 	return opts.DeleteBranch, nil
 }
 
-func confirmSurvey() (bool, error) {
-	var confirm bool
-	submit := &survey.Confirm{
-		Message: "Submit?",
-		Default: true,
+func confirmSurvey(allowEditMsg bool) (shared.Action, error) {
+	const (
+		submitLabel        = "Submit"
+		editCommitMsgLabel = "Edit commit message"
+		cancelLabel        = "Cancel"
+	)
+
+	options := []string{submitLabel}
+	if allowEditMsg {
+		options = append(options, editCommitMsgLabel)
+	}
+	options = append(options, cancelLabel)
+
+	var result string
+	submit := &survey.Select{
+		Message: "What's next?",
+		Options: options,
+	}
+	err := prompt.SurveyAskOne(submit, &result)
+	if err != nil {
+		return shared.CancelAction, fmt.Errorf("could not prompt: %w", err)
+	}
+
+	switch result {
+	case submitLabel:
+		return shared.SubmitAction, nil
+	case editCommitMsgLabel:
+		return shared.EditCommitMessageAction, nil
+	default:
+		return shared.CancelAction, nil
+	}
+}
+
+func commitMsgSurvey(editorCommand string) (string, error) {
+	var result string
+	q := &surveyext.GhEditor{
+		EditorCommand: editorCommand,
+		Editor: &survey.Editor{
+			Message:  "Body",
+			FileName: "*.md",
```

Comment:
> For squash merge, the default commit message is the summary of all commits that were originally in the pull request. The latter case might be tricky to generate locally, so we could just leave this as-is.

The use case described in #1023 states that the user may edit the default commit message in their editor, that's why I first added it (the graphql api provides it) if no body was provided with `--body`. That way it would work the same way GitHub does in web.

I see Sam removed that part, do you think it's a valid use case for re-adding it again?.

Reply:
I was unable to get the pre-population to work properly for squash merges which is why I removed it completely. The `viewerMergeBody` GraphQL field on `PullRequests` returned only blank strings in the couple of tests I did, so I thought it better to leave it to be unimplemented than partially working.